Source/CarthageKit/Project.swift

```diff
@@ -354,6 +354,45 @@ public final class Project {
 			.collect()
 			.map(ResolvedCartfile.init)
 	}
+	
+	/// Attempts to determine which of the project's Carthage
+	/// dependencies are out of date.
+	///
+	/// This will fetch dependency repositories as necessary, but will not check
+	/// them out into the project's working directory.
+	public func outdatedDependencies(includeTransientDependencies: Bool) -> SignalProducer<[Dependency<OutdatedVersion>], CarthageError> {
+		let currentDependenciesProducer = loadResolvedCartfile()
+			.map { $0.dependencies }
+		let updatedDependenciesProducer = updatedResolvedCartfile()
+			.map { $0.dependencies }
+		let outdatedDependenciesProducer = combineLatest(currentDependenciesProducer, updatedDependenciesProducer)
+			.map { (currentDependencies, updatedDependencies) -> [Dependency<OutdatedVersion>] in
+				var outdatedDependencies: [Dependency<OutdatedVersion>] = []
+
+				for updatedDependency in updatedDependencies {
+					if let resolvedDependency = currentDependencies.filter({ $0.project == updatedDependency.project }).first where resolvedDependency.version != updatedDependency.version {
```

Comment:
Rather than filtering the dependencies on each iteration, could you build a `[Project: Dependency<PinnedVersion>]` upfront and then check `let resolvedDependency = currentDependencies[updatedDependency.project]`?